Assignment2/kaprekarConstant.js

```diff
@@ -0,0 +1,45 @@
+let num = "2111";
+let iterationCount = 0;
+
+kaprekarRoutine(num);
+
+function isFourDigit(num) {
+  return num.length === 4;
+}
+
+function addLeadingZero(num) {
+  return "0" + num;
+}
+
+function convertIntoDescending(num) {
+  let digitsArray = num.split("");
+  digitsArray.sort();
+  digitsArray.reverse();
+  return digitsArray.join("");
+}
+function convertIntoAscending(num) {
+  let digitsArray = num.split("");
+  digitsArray.sort();
+  return digitsArray.join("");
+}
+
+function kaprekarRoutine(num) {
+  iterationCount++;
+  if (!isFourDigit(num)) {
+    num = addLeadingZero(num);
```

Comment:
this function is only useful if length of num is 3, if its <3, it will only ass single 0